app/models/user.rb

```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```

Comment:
- trước dấu` }` và sau dấu` {` không có khoảng trắng
- `downcase` vs `downcase!` khác nhau ntn?

Reply:
downcase trả về 1 bản sao của string vs các chữ hoa thay băng chữ thường
downcase! đưa nội dung của string về nil

Reply:
downcase trả về bản copy của string vs các ký tự in hoa chuyển thanh thường
downcase! đưa nội dung của string về nil